File: packages/sdk-kotlin/src/main/kotlin/org/melaya/AssistantAPI.kt
package org.melaya

import org.json.JSONObject

/**
 * Assistant API — get and save the caller's onboarding / persona profile.
 *
 * The profile is stored envelope-encrypted (service=`assistant_profile`) and used
 * to personalise the Melaya AI assistant experience.
 *
 * Paths:
 *   - `GET /api/v1/private/assistant/profile` — get assistant profile
 *   - `PUT /api/v1/private/assistant/profile` — save assistant profile
 *
 * @example
 * ```kotlin
 * val profile = melaya.assistant.getProfile()
 * melaya.assistant.setProfile(mapOf("name" to "Alex", "goals" to listOf("grow my trading edge")))
 * ```
 */
class AssistantAPI internal constructor(private val http: HttpClient) {

    /** Get the caller's assistant onboarding profile (decrypted from service=`assistant_profile`). */
    fun getProfile(): JSONObject {
        return http.get("/api/v1/private/assistant/profile").asObject()
    }

    /**
     * Save the caller's assistant onboarding profile.
     *
     * @param profile A map with optional keys: `name`, `goals` (list), `context`, `preferences` (map).
     */
    fun setProfile(profile: Map<String, Any?>): JSONObject {
        return http.put("/api/v1/private/assistant/profile", profile).asObject()
    }
}

File: packages/sdk-csharp/Melaya/AssistantApi.cs
namespace Melaya;

/// <summary>
/// Assistant API — get and save the caller's onboarding / persona profile.
/// Maps to <c>/api/v1/private/assistant/profile</c>.
/// The profile is stored envelope-encrypted (service=<c>assistant_profile</c>)
/// and used to personalise the in-app Assistant experience.
/// </summary>
/// <example>
/// <code>
/// var profile = await m.Assistant.GetProfileAsync();
/// await m.Assistant.SetProfileAsync(new AssistantProfile { Name = "Alex", Goals = new[] { "grow my trading edge" }.ToList() });
/// </code>
/// </example>
public sealed class AssistantApi
{
    private readonly MelayaHttpClient _http;

    internal AssistantApi(MelayaHttpClient http) => _http = http;

    /// <summary>Get the caller's assistant onboarding profile.</summary>
    public async Task<AssistantProfile> GetProfileAsync(CancellationToken ct = default)
    {
        return await _http.GetAsync<AssistantProfile>("/api/v1/private/assistant/profile", ct: ct).ConfigureAwait(false);
    }

    /// <summary>Save the caller's assistant onboarding profile.</summary>
    public async Task<BoolResult> SetProfileAsync(AssistantProfile profile, CancellationToken ct = default)
    {
        return await _http.PutAsync<BoolResult>("/api/v1/private/assistant/profile", profile, ct).ConfigureAwait(false);
    }
}
